src/meter_set_numeric.lua

```diff
@@ -0,0 +1,389 @@
+            meters[i] = user_entry[i]:GetInteger()
+        end
+    end
+    return ok
+end
+
+function encode_existing_meter(time_sig, meters, index_offset)
+    -- convert existing meter to our { meters } filing method
+    if time_sig.CompositeTop then
+        local comp_top = time_sig:CreateCompositeTop()
+        local group_count = comp_top:GetGroupCount()
+        if group_count > 3 then 
+            group_count = 3 -- three composites max
+        end
+        for i = 0, group_count - 1 do -- add elements of each group
+            local top_string = ""  -- empty string to index_offset
+            for element_count = 0, comp_top:GetGroupElementCount(i) - 1 do
+                if element_count > 0 then -- plus sign after first number
+                    top_string = top_string .. "+" 
+                end
+                top_string = top_string .. comp_top:GetGroupElementBeats(i, element_count)
+            end
+            meters[(i * 2) + index_offset] = top_string -- save the string result
+        end
+    else    -- non-composite TOP, simple number string
+        meters[index_offset] = "" .. time_sig.Beats
+    end
+
+    if time_sig.CompositeBottom then
+        local comp_bottom = time_sig:CreateCompositeBottom()
+        local group_count = comp_bottom:GetGroupCount()
+        if group_count > 3 then
+            group_count = 3 -- 3 composites max
+        end
+        for i = 0, group_count - 1 do
+            -- assume one element for each composite bottom
+            local beat_duration = comp_bottom:GetGroupElementBeatDuration(i, 0)
+            local bottom_offset = (i * 2) + index_offset -- offset for individual composites
+            if beat_duration % 3 == 0 and not string.find(meters[bottom_offset], "+") then 
+                meters[bottom_offset] = "" .. ( meters[bottom_offset] * 3 ) -- create compound meter
+                beat_duration = beat_duration / 3
+            end
+            meters[bottom_offset + 1] = math.floor(4096 / beat_duration) -- round to integer in case
+        end
+    else    -- non-composite BOTTOM
+        local beat_duration = time_sig.BeatDuration
+        if beat_duration % 3 == 0 and not string.find(meters[index_offset], "+") then
+            meters[index_offset] = "" .. ( meters[index_offset] * 3 ) -- create compound meter
+            beat_duration = beat_duration / 3
+        end
+        meters[index_offset + 1] = math.floor(4096 / beat_duration) -- round to integer in case
+    end
+end
+
+function copy_meters_from_score(meters, measure_number)
+    local measure = finale.FCMeasure()
+    measure:Load(measure_number)
+    encode_existing_meter(measure:GetTimeSignature(), meters, 1)  -- encode primary meter - offset to meters[1]
+    if measure.UseTimeSigForDisplay then    -- encode DISPLAY meter - offset to meters[7]
+        encode_existing_meter(measure.TimeSignatureForDisplay, meters, 7)
+    end
+end
+
+function is_positive_integer(x)
```

Comment:
In general, it's good coding practice that functions that start with `is` return booleans. Just makes things more readable.

So here I'd return a boolean:

```lua
function is_positive_integer(x)
    return x ~= nil and x > 0 and x % 1 == 0
end
```

In case you're not familiar with the `%` operator, it's called the modulo operator. Essentially, it takes the remainder after doing division.

```
print(18 % 10) -- 8
print(5 % 2) -- 1
print(1.5 % 1) -- 0.5
```